src/main/java/com/couchbase/lite/javascript/ReplicationFilterBlockRhino.java

```diff
@@ -52,28 +58,27 @@ public ReplicationFilterBlockRhino(String src){
 
     @Override
     public boolean filter(SavedRevision revision, Map<String, Object> params) {
-        org.mozilla.javascript.Context ctx = org.mozilla.javascript.Context.enter();
-        try {
-            ctx.setOptimizationLevel(-1);
-            ctx.setWrapFactory(wrapFactory);
-
-            Scriptable localScope = ctx.newObject(scope);
-            localScope.setPrototype(scope);
-            localScope.setParentScope(null);
-
-            Object jsDocument = org.mozilla.javascript.Context.javaToJS(revision.getProperties(), localScope);
-            Object jsParams = org.mozilla.javascript.Context.javaToJS(params, localScope);
-
+        synchronized (lockFunction) {
```

Comment:
I WAS NOT sure if synchronized `Function.call()` is enough. According to my test results, synchronized `Function.call()` is enough. I will update PR.